src/Dev/CsvBulkLoader.php

```diff
@@ -388,4 +394,22 @@ public function hasHeaderRow()
     {
         return ($this->hasHeaderRow || isset($this->columnMap));
     }
+
+    /**
+     * Cast $val to the appropriate type for the given field it is being imported into
+     */
+    private function castVal(mixed $val, string $class, string $fieldName, DataObjectSchema $schema): mixed
+    {
+        $spec = $schema->fieldSpec($class, $fieldName);
+        if (!$spec) {
+            return $val;
+        }
+        $dbField = DBField::create_field($spec, null);
+        if (is_a($dbField, DBInt::class)) {
+            $val = (int) $val;
+        } elseif (is_a($dbField, DBFloat::class) || is_a($dbField, DBDecimal::class)) {
+            $val = (float) $val;
+        }
+        return $val;
+    }
```

Comment:
`DBInt` does this cast already in `setValue()`.
For consistency, the other numeric `DBField` classes should cast numeric strings to the correct type. We can then remove this code.

Reply:
Done. DBDecimal already did this so I simply copied the setValue() code from DBDecimal to DBFloat. I also copied DBDecimalTest.php to DBFloatTest.php and adatped to use DBFloat. Have deleted the castValue() code form CsvBulkLoader though kept the changes in CsvBulkLoaderTest